docs/upgrade/11_to_12/workflows_mysql5.py

```diff
@@ -0,0 +1,425 @@
+execute_query(connection, delete_workflow_operation_configuration_table)
+
+## Create new tables
+#  Currently added indexes:
+#  - Indexes for bidirectional relationships between Workflow <-> Operation <-> Configuration
+#  - mediaPackageId, seriesId for oc_workflow
+print("Create tables...")
+create_workflow_table = f"""
+CREATE TABLE IF NOT EXISTS {workflow_table_name} (
+  id BIGINT(20),
+  state INT(11),
+  template VARCHAR(255),
+  title VARCHAR(255),
+  description VARCHAR(255),
+  parent BIGINT(20),
+  creatorId VARCHAR(255),
+  organizationId VARCHAR(255),
+  dateCreated DATETIME,
+  dateCompleted DATETIME,
+  mediaPackage LONGTEXT,
+  mediaPackageId VARCHAR(128),
+  seriesId VARCHAR(128),
+  PRIMARY KEY (id),
+  INDEX (mediaPackageId),
+  INDEX (seriesId)
+) ENGINE = InnoDB
+"""
+
+create_workflow_configuration_table = f"""
+CREATE TABLE IF NOT EXISTS {workflow_configuration_table_name} (
+  workflow_id BIGINT(20),
+  key_part VARCHAR(255) NOT NULL,
+  value_part LONGTEXT,
+  INDEX (workflow_id)
+) ENGINE = InnoDB
+"""
+
+# ??? Missing "executionHistory" maybe?
+create_workflow_operation_table = f"""
+CREATE TABLE IF NOT EXISTS {workflow_operation_table_name} (
+  id BIGINT(20) AUTO_INCREMENT,
+  template VARCHAR(255),
+  job BIGINT(20),
+  state INT(11),
+  description VARCHAR(255),
+  holdurl VARCHAR(255),
+  holdActionTitle VARCHAR(255),
+  failOnError TINYINT(1),
+  if_condition VARCHAR(255),
+  unless_condition VARCHAR(255),
+  exceptionHandlerWorkflow VARCHAR(255),
+  abortable TINYINT(1) DEFAULT 0,
+  continuable TINYINT(1) DEFAULT 0,
+  started DATETIME,
+  completed DATETIME,
+  timeInQueue BIGINT(20),
+  maxAttempts INT(11),
+  failedAttempts INT(11) DEFAULT 0,
+  executionHost VARCHAR(255),
+  retryStrategy INT(11),
+  POSITION INT(11),
+  INSTANCE_id BIGINT(20),
+  operations_ORDER INT(11),
+  PRIMARY KEY (id),
+  INDEX (INSTANCE_id)
+) ENGINE = InnoDB
+"""
+
+create_workflow_operation_configuration_table = f"""
+CREATE TABLE IF NOT EXISTS {workflow_operation_configuration_table_name} (
+  workflow_operation_id BIGINT(20),
+  key_part VARCHAR(255) NOT NULL,
+  value_part LONGTEXT,
+  INDEX (workflow_operation_id)
+) ENGINE = InnoDB
+"""
+
+execute_query(connection, create_workflow_table)
+execute_query(connection, create_workflow_configuration_table)
+execute_query(connection, create_workflow_operation_table)
+execute_query(connection, create_workflow_operation_configuration_table)
+
+### Get information from database
+print("Collect information from oc_job table...")
+select_payload_from_job_table = """
+SELECT payload FROM oc_job WHERE operation="START_WORKFLOW"
+"""
+select_date_created_from_job_table = """
+SELECT date_created FROM oc_job WHERE operation="START_WORKFLOW"
+"""
+select_date_completed_from_job_table = """
+SELECT date_completed FROM oc_job WHERE operation="START_WORKFLOW"
+"""
+
+payloads = execute_read_query(connection, select_payload_from_job_table)
```

Comment:
This will potentially load tens of thousands of workflows at the same time from the database. Not a good idea. Load them in chunks of maybe 100 workflows at a time.